common/src/main/scala/explore/common/SSOClient.scala

```diff
@@ -45,17 +48,37 @@ object SSOClient {
       window.location.href = uri"$baseSSO/auth/v1/stage1?state=$returnUrl".toString
     }
 
-  def whoami[F[_]: Sync](
+  def vault[F[_]: ConcurrentEffect: Logger](
```

Comment:
Do we need the power of a ConcurrentEffect here? Are we cancelling?

Reply:
The creation of a `Deferred` needs `Concurrent` and we also need `Effect` to call `runAsyncCB`